executor/src/witgen/machines/double_sorted_witness_machine.rs

```diff
@@ -77,20 +78,27 @@ impl<T: FieldElement> DoubleSortedWitnesses<T> {
         ]
         .into_iter()
         .collect();
-        if expected_witnesses
+        let difference = expected_witnesses
             .symmetric_difference(&columns)
-            .next()
-            .is_none()
-        {
-            Some(Self {
-                name,
-                namespace,
-                degree: fixed_data.degree,
-                ..Default::default()
-            })
-        } else {
-            None
+            .collect::<HashSet<_>>();
+        match difference.len() {
+            0 => Some(false),
+            2 => {
+                if difference == ["m_diff_upper", "m_diff_lower"].iter().collect() {
+                    Some(true)
+                } else {
+                    None
+                }
+            }
+            _ => None,
         }
```

Comment:
This whole `match` looks a big magic, what is it doing? Why are the arms 0 and 2?